src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -56,6 +56,22 @@ private string ValueToString()
             };
         }
 
+        private static bool ValueToString<TEnum>(TEnum value, Span<char> destination, out int charsWritten, bool isHexFormat = false) where TEnum : struct, Enum
+        {
+            return (Type.GetTypeCode(typeof(TEnum))) switch
+            {
+                TypeCode.SByte => Unsafe.As<TEnum, sbyte>(ref value).TryFormat(destination, out charsWritten, isHexFormat ? "X2" : null),
+                TypeCode.Byte => Unsafe.As<TEnum, byte>(ref value).TryFormat(destination, out charsWritten, isHexFormat ? "X2" : null),
+                TypeCode.Int16 => Unsafe.As<TEnum, short>(ref value).TryFormat(destination, out charsWritten, isHexFormat ? "X4" : null),
+                TypeCode.UInt16 => Unsafe.As<TEnum, ushort>(ref value).TryFormat(destination, out charsWritten, isHexFormat ? "X4" : null),
+                TypeCode.UInt32 => Unsafe.As<TEnum, uint>(ref value).TryFormat(destination, out charsWritten, isHexFormat ? "X8" : null),
+                TypeCode.Int32 => Unsafe.As<TEnum, int>(ref value).TryFormat(destination, out charsWritten, isHexFormat ? "X8" : null),
+                TypeCode.UInt64 => Unsafe.As<TEnum, ulong>(ref value).TryFormat(destination, out charsWritten, isHexFormat ? "X16" : null),
+                TypeCode.Int64 => Unsafe.As<TEnum, long>(ref value).TryFormat(destination, out charsWritten, isHexFormat ? "X16" : null),
+                _ => throw new InvalidOperationException(SR.InvalidOperation_UnknownEnumType),
```

Comment:
Technically, this case should never happen, right? What do you think about an `UnreachableException`.

As far as I know, this is currently only for static code analysis, but perhaps JIT will take this into account in the future and optimize it accordingly.

Reply:
Good idea, updated

Reply:
This case can happen for enums that are backed by floats, doubles, etc. These enums are not expressible in C#, but they are supported by the runtime for compat. Non-standard enums like that are sometimes used by obfuscators.

Reply:
Also, notice that the non-generic ValueToString handles these cases.

Reply:
Thank you for clarification @jkotas 
@heathbm Can you please change it back to InvalidOperationException

Reply:
@deeprobin Sounds good, I have revered it back